src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherV3Test.java

```diff
@@ -0,0 +1,286 @@
+    DataUseV3 purpose = new DataUseBuilderV3().setMethodsResearch(true).build();
+    assertDeny(purpose, dataset);
+  }
+
+  @Test
+  public void testPOA_positive_case_1() {
+    DataUseV3 dataset = new DataUseBuilderV3().setGeneralUse(true).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setPopulationOriginsAncestry(true).build();
+    assertApprove(purpose, dataset);
+  }
+
+  @Test
+  public void testPOA_positive_case_2() {
+    DataUseV3 dataset = new DataUseBuilderV3().setPopulationOriginsAncestry(true).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setPopulationOriginsAncestry(true).build();
+    assertApprove(purpose, dataset);
+  }
+
+  @Test
+  public void testPOA_negative() {
+    DataUseV3 dataset = new DataUseBuilderV3().setHmbResearch(true).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setPopulationOriginsAncestry(true).build();
+    assertDeny(purpose, dataset);
+  }
+
+  @Test
+  public void testCommercial_positive_case_1() {
+    DataUseV3 dataset = new DataUseBuilderV3().setCommercialUse(true).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setCommercialUse(true).build();
+    assertApprove(purpose, dataset);
+  }
+
+  @Test
+  public void testCommercial_positive_case_2() {
+    DataUseV3 dataset = new DataUseBuilderV3().setGeneralUse(true).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setCommercialUse(true).build();
+    assertApprove(purpose, dataset);
+  }
+
+  @Test
+  public void testCommercial_positive_case_3() {
+    DataUseV3 dataset = new DataUseBuilderV3().setNonProfitUse(null).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setCommercialUse(true).build();
+    assertApprove(purpose, dataset);
+  }
+
+  @Test
+  public void testCommercial_negative_case_1() {
+    DataUseV3 dataset = new DataUseBuilderV3().setNonProfitUse(true).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setCommercialUse(true).build();
+    assertDeny(purpose, dataset);
+  }
+
+  @Test
+  public void testCommercial_negative_case_2() {
+    DataUseV3 dataset = new DataUseBuilderV3().setCommercialUse(false).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setCommercialUse(true).build();
+    assertDeny(purpose, dataset);
+  }
+
+  @Test
+  public void testHMB_positive_case_1() {
+    DataUseV3 dataset = new DataUseBuilderV3().setGeneralUse(true).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setHmbResearch(true).build();
+    assertApprove(purpose, dataset);
+  }
+
+  @Test
+  public void testHMB_positive_case_2() {
+    DataUseV3 dataset = new DataUseBuilderV3().setHmbResearch(true).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setHmbResearch(true).build();
+    assertApprove(purpose, dataset);
+  }
+
+  @Test
+  public void testHMB_negative() {
+    DataUseV3 dataset = new DataUseBuilderV3().setHmbResearch(true).build();
+    DataUseV3 purpose = new DataUseBuilderV3().setGeneralUse(true).build();
+    assertDeny(purpose, dataset);
+  }
```

Comment:
There should be two more negative HMB purpose cases according to the algorithm doc:

> Deny if the dataset's Primary DUO terms are DS-, POA

I wrote this to convince myself that the logic was working as expected. I leave it to you to organize as you see fit.
```
  @Test
  public void testMatchHMB() {
    DataUseV3 purpose = new DataUseBuilderV3().setHmbResearch(true).build();
    DataUseV3 dataset = new DataUseBuilderV3().setHmbResearch(true).build();
    assertApprove(purpose, dataset);

    purpose = new DataUseBuilderV3().setHmbResearch(true).build();
    dataset = new DataUseBuilderV3().setGeneralUse(true).build();
    assertApprove(purpose, dataset);

    purpose = new DataUseBuilderV3().setHmbResearch(true).build();
    dataset = new DataUseBuilderV3().setDiseaseRestrictions(Collections.singletonList(cancerNode)).build();
    assertDeny(purpose, dataset);

    purpose = new DataUseBuilderV3().setHmbResearch(true).build();
    dataset = new DataUseBuilderV3().setPopulationOriginsAncestry(true).build();
    assertDeny(purpose, dataset);
  }
```

Reply:
Thank you!